homeassistant/components/zwave_mqtt/services.py

```diff
@@ -0,0 +1,173 @@
+"""Methods and classes related to executing Z-Wave commands and publishing these to hass."""
+import logging
+
+from openzwavemqtt.const import CommandClass, ValueType
+import voluptuous as vol
+
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from . import const
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ZWaveServices:
+    """Class that holds our services ( Zwave Commands) that should be published to hass."""
+
+    def __init__(self, hass, manager, data_nodes):
+        """Initialize with both hass and ozwmanager objects."""
+        self._hass = hass
+        self._manager = manager
+        self._data_nodes = data_nodes
+
+    @callback
+    def register(self):
+        """Register all our services."""
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_ADD_NODE,
+            self.add_node,
+            schema=vol.Schema(
+                {
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                    vol.Optional(const.ATTR_SECURE, default=False): vol.Coerce(bool),
+                }
+            ),
+        )
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_REMOVE_NODE,
+            self.remove_node,
+            schema=vol.Schema(
+                {vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int)}
+            ),
+        )
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_REMOVE_FAILED_NODE,
+            self.remove_failed_node,
+            schema=vol.Schema(
+                {
+                    vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                }
+            ),
+        )
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_REPLACE_FAILED_NODE,
+            self.replace_failed_node,
+            schema=vol.Schema(
+                {
+                    vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                }
+            ),
+        )
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_CANCEL_COMMAND,
+            self.cancel_command,
+            schema=vol.Schema(
+                {vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int)}
+            ),
+        )
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_SET_CONFIG_PARAMETER,
+            self.set_config_parameter,
+            schema=vol.Schema(
+                {
+                    vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
```

Comment:
Could we validate this with live data since we have access to the manager here ?

Reply:
We will rework the services in a follow up PR. For now we've removed all services except add and remove node, which we need to ease testing.